File: space-game/Core/Helpers/PathHelper.vb
Imports System.Text

Namespace Helpers
    Public Class PathHelper

        Public Shared Function SitePathCombine(path1 As String, path2 As String) As String

            If {path1, path2}.Any(Function(p) String.IsNullOrWhiteSpace(p)) Then
                Throw New ArgumentNullException()
            End If

            dim result as New StringBuilder

            result.Append(path1.TrimEnd("/"c))
            result.Append("/")
            result.Append(path2.TrimStart("/"c))

            Return result.ToString()
        End Function

    End Class
End Namespace

File: space-game/Core/Helpers/PathHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Core.Helpers
{
    public class PathHelper
    {
        public static string SitePathCombine(string path1, string path2)
        {

            if (new List<string> () { 
                path1,
				path2
            }.Any(p => string.IsNullOrWhiteSpace(p))) {
                throw new ArgumentNullException();
            }

            StringBuilder result = new StringBuilder();

            result.Append(path1.TrimEnd('/'));
            result.Append("/");
            result.Append(path2.TrimStart('/'));

            return result.ToString();
        }

    }
}
